migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,210 @@
+		return &storage.ReportSnapshot{
+			ReportConfigurationId: v2Config.GetId(),
+			Name:                  v2Config.GetName(),
+			Description:           v2Config.GetDescription(),
+			Type:                  storage.ReportSnapshot_VULNERABILITY,
+			ReportId:              uuid.NewV4().String(),
+			Collection: &storage.CollectionSnapshot{
+				Id: v2Config.GetResourceScope().GetCollectionId(),
+			},
+			Schedule: v2Config.GetSchedule(),
+			ReportStatus: &storage.ReportStatus{
+				RunState:          storage.ReportStatus_DELIVERED,
+				ReportRequestType: storage.ReportStatus_SCHEDULED,
+				CompletedAt:       v1Config.LastSuccessfulRunTime,
+			},
+		}
+	}
+	return nil
+}
+
+func migrate(database *types.Databases) error {
+	db := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt)
+	db = db.WithContext(database.DBCtx).Table("report_configurations")
+
+	dbSnapshot := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, dbSnapshot, updatedSchema.CreateTableReportSnapshotsStmt)
+	dbSnapshot = dbSnapshot.WithContext(database.DBCtx).Table("report_snapshots")
+
+	dbNotifier := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, dbNotifier, updatedSchema.CreateTableReportConfigurationsStmt.Children[0])
+	dbNotifier = dbNotifier.WithContext(database.DBCtx).Table("report_configurations_notifiers")
+
+	query := db.WithContext(database.DBCtx).Table(updatedSchema.ReportConfigurationsTableName).Select("serialized")
+	rows, err := query.Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to iterate table %s", "report_configurations")
+	}
+	defer func() { _ = rows.Close() }()
+	var convertedReportConfigs []*updatedSchema.ReportConfigurations
+	var reportSnapshots []*updatedSchema.ReportSnapshots
+	var reportNotifiers []*updatedSchema.ReportConfigurationsNotifiers
+	for rows.Next() {
+		if rows.Err() != nil {
+			utils.Should(rows.Err())
+			return errors.Wrapf(rows.Err(), "failed to get rows for %s", "report_configurations")
+		}
+		var reportConfig *updatedSchema.ReportConfigurations
+		if err = db.ScanRows(rows, &reportConfig); err != nil {
+			return errors.Wrap(err, "failed to scan rows")
+		}
+		// convert to report config proto
+		reportConfigProto, err := updatedSchema.ConvertReportConfigurationToProto(reportConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert %+v to proto", reportConfigProto)
+		}
+		// skip if version=1 since config is migrated
+		// skip if version=2
+		if reportConfigProto.Version != 0 {
+			continue
+		}
+		if reportConfigProto.Version == 0 && reportConfigProto.GetResourceScope() != nil {
+			reportConfigProto.Version = 2
+			// convert report config proto back to gorm model
+			convertedGormConfig, err := updatedSchema.ConvertReportConfigurationFromProto(reportConfigProto)
+			if err != nil {
+				return errors.Wrapf(err, "failed to convert from proto %+v", reportConfigProto)
```

Comment:
I think we should still try to migrate other configs if some error out. Add errors to `errorList` for configs that fail to migrate, and continue. Log all the errors at the end of migration.